packages/push/src/genome/plushy.rs

```diff
@@ -72,63 +138,66 @@ mod test {
     #[test]
     #[allow(clippy::unwrap_used)]
     fn generator() {
-        let instructions = vec![
-            PushInstruction::IntInstruction(IntInstruction::Add),
-            PushInstruction::IntInstruction(IntInstruction::Subtract),
-            PushInstruction::IntInstruction(IntInstruction::Multiply),
-            PushInstruction::IntInstruction(IntInstruction::ProtectedDivide),
+        let instructions: Vec<PushInstruction> = vec![
+            IntInstruction::Add.into(),
+            IntInstruction::Subtract.into(),
+            IntInstruction::Multiply.into(),
+            IntInstruction::ProtectedDivide.into(),
         ];
+        let gene_generator = GeneGenerator::with_uniform_close_probability(instructions);
         let mut rng = thread_rng();
         let plushy: Plushy = CollectionGenerator {
             size: 10,
-            element_generator: instructions,
+            element_generator: gene_generator,
         }
         .generate(&mut rng)
         .unwrap();
-        assert_eq!(10, plushy.instructions.len());
+        assert_eq!(10, plushy.genes.len());
     }
 
     #[test]
     fn umad() {
         let mut rng = thread_rng();
 
-        let instruction_options = [PushInstruction::InputVar(VariableName::from("x"))];
+        let instruction_options = [PushGene::Instruction(PushInstruction::InputVar(
+            VariableName::from("x"),
+        ))];
         let umad = Umad::new(0.3, 0.3, instruction_options);
 
-        let parent_instructions = vec![
-            PushInstruction::IntInstruction(IntInstruction::Add),
-            PushInstruction::BoolInstruction(BoolInstruction::And),
-            PushInstruction::BoolInstruction(BoolInstruction::Or),
-            PushInstruction::IntInstruction(IntInstruction::Multiply),
-            PushInstruction::IntInstruction(IntInstruction::Add),
-            PushInstruction::BoolInstruction(BoolInstruction::And),
-            PushInstruction::BoolInstruction(BoolInstruction::Or),
-            PushInstruction::IntInstruction(IntInstruction::Multiply),
-            PushInstruction::IntInstruction(IntInstruction::Add),
-            PushInstruction::BoolInstruction(BoolInstruction::And),
-            PushInstruction::BoolInstruction(BoolInstruction::Or),
-            PushInstruction::IntInstruction(IntInstruction::Multiply),
-            PushInstruction::IntInstruction(IntInstruction::Add),
-            PushInstruction::BoolInstruction(BoolInstruction::And),
-            PushInstruction::BoolInstruction(BoolInstruction::Or),
-            PushInstruction::IntInstruction(IntInstruction::Multiply),
-            PushInstruction::IntInstruction(IntInstruction::Add),
-            PushInstruction::BoolInstruction(BoolInstruction::And),
-            PushInstruction::BoolInstruction(BoolInstruction::Or),
-            PushInstruction::IntInstruction(IntInstruction::Multiply),
+        let parent_genes: Vec<PushGene> = vec![
```

Comment:
```suggestion
        let genes: Vec<PushGene> = vec![
```
and then we can just use `genes` and not `genes: parent_genes` below